server/src/test/java/com/linecorp/centraldogma/server/authentication/ReplicatedLoginAndLogoutTest.java

```diff
@@ -92,9 +92,11 @@ public void setup() throws Exception {
                 1, new ZooKeeperAddress("127.0.0.1", zkQuorumPort1, zkElectionPort1, zkClientPort1),
                 2, new ZooKeeperAddress("127.0.0.1", zkQuorumPort2, zkElectionPort2, zkClientPort2));
 
+        final AuthenticationProviderFactory factory = new TestAuthenticationProviderFactory();
+
         replica1 = new CentralDogmaBuilder(tempDir.newFolder())
                 .port(port1, SessionProtocol.HTTP)
-                .securityConfig(newSecurityConfig())
+                .authenticationProviderFactory(factory)
```

Comment:
Question: How would you let a user choose a provider factory and provide a non-standard parameters to it?

Reply:
For example, our server distribution will contains more than one authentication providers (e.g. SAML and Shiro). How would a user be able to specify an authentication provider and its configuration parameters? For Shiro, we could externalize the configuration to an ini file, but perhaps we can keep all parameters in dogma.json?

Reply:
I assumed that a user should add only a single authentication provider to the classpath, so I limited the number of providers to 1. https://github.com/line/centraldogma/pull/279/files#diff-6c1d7419420eb829a077a6ce7de69906R200

Also, providing non-standard parameters can be performed by passing a `File` which is specified by `-securityConfig` parameter.

For now, I think it's enough for our usage, but we might need to think again when adding SAML auth provider. Please let me know if you have a concern about the current way. :-)

Reply:
Wouldn't that complicate the setup process? We will not be able to provide a tarball that support all authentication providers. How about using some reflection-based way? e.g.

```json
{
    "authentication": {
        "type": "<FQCN of the auth provider factory>",
        // ... the properties that are going to be used by the auth provider ...
    }
}
```

I think this doesn't have to be part of this pull request, though.

Reply:
I see. It makes sense. I agree that reflection-based way would be simpler than the current way. Will fix it and check `dist` module to confirm that the auth extensions are copied into `lib`. Thanks! :-)

Reply:
Thank you!